nerfacc/estimators/occ_grid.py

```diff
@@ -258,6 +258,80 @@ def update_every_n_steps(
                 warmup_steps=warmup_steps,
             )
 
+    @torch.no_grad()
+    def mark_invisible_cells(
+        self,
+        K: Tensor,
+        c2w: Tensor,
+        width: int,
+        height: int,
+        near_plane: float = 0.0,
+        chunk: int = 32**3,
+    ) -> None:
+        """Mark the cells that aren't covered by the cameras with density -1.
+        Should only be executed once before training starts.
+
+        Note:
+            This code is adapted from: https://github.com/kwea123/ngp_pl/blob/master/models/networks.py
+
+        Args:
+            K: Camera intrinsics of shape (N, 3, 3) or (1, 3, 3).
+            c2w: Camera to world poses of shape (N, 3, 4) or (N, 4, 4).
+            width: Image width in pixels
+            height: Image height in pixels
+            near_plane: Near plane distance
+            chunk: The chunk size to split the cells (to avoid OOM)
+        """
+        assert K.dim() == 3 and K.shape[1:] == (3, 3)
```

Comment:
Should we ensure that the `c2w` and `K` share the same value for `N`?

Reply:
Will add